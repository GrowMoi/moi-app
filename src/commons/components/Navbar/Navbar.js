import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import { connect } from 'react-redux';
import { Image, StatusBar, View } from 'react-native';
import styled from 'styled-components/native';
import navbarImageLandscape from '../../../../assets/images/navbar/barra_sup_landscape.png';
import navbarImagePortrait from '../../../../assets/images/navbar/barra_superior_portrait.png';
import { Size } from '../../styles';
import { LANDSCAPE } from '../../../constants';

const Bar = styled(View)`
  position: absolute;
  left: 0;
  right: 0;
  top: -2;
  height: ${Size.navbarHeight};
`;

const BackgroundImage = styled(Image)`
  position: relative;
  flex: 1;
  width: ${props => props.width};
  height: ${Size.navbarHeight};
`;

const mapStateToProps = state => ({
  device: state.device,
});

@connect(mapStateToProps)
class Navbar extends Component {
  static propTypes = {
    device: PropTypes.object,
  }

  goToProfile = () => {
    Actions.profile();
  }

  render() {
    const { device } = this.props;
    const { width, orientation } = device.dimensions;
    const currentImage = orientation === LANDSCAPE ? navbarImageLandscape : navbarImagePortrait;

    return (
      <Bar>
        <StatusBar hidden />
        <BackgroundImage
          resizeMode='stretch'
          width={width}
          source={currentImage}
        />
      </Bar>
    );
  }
}

export default Navbar;
