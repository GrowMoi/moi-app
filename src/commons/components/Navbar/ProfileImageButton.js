import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Actions } from 'react-native-router-flux';
import { Image, TouchableOpacity } from 'react-native';
import MoIcon from '../MoIcon/MoIcon';
import withSound from '../../utils/withSound';
import { Size } from '../../styles';
import ProfileAvatar from '../Profile/Profile'

class ProfileImageButton extends Component {

  renderProfileIconWithSound = (profile) => {
    const TouchableOpacityWithSound = withSound(TouchableOpacity);
    return (
      <TouchableOpacityWithSound
        soundName="profile"
        onPress={() => Actions.profile()}
      >
        <ProfileAvatar width={Size.profileAvatarSize} userImageUri={profile.image || profile.avatar}
        />
      </TouchableOpacityWithSound>
    );
  }

  render() {
    const { profile } = this.props;
    return this.renderProfileIconWithSound(profile);
  }
}

const mapStateToProps = state => ({
  profile: state.user.profile
})

export default connect(mapStateToProps)(ProfileImageButton)
