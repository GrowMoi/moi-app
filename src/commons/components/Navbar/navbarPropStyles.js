import React from 'react';
import { Icon } from 'expo';
import { Platform, Dimensions, TouchableOpacity } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { Font } from '../../styles';
import styled from 'styled-components/native';
import ProfileImageButton from './ProfileImageButton';
import { Size } from '../../styles';

//REDUX
import store from '../../../store'
import * as sideActions from '../../../actions/sideActions'

const { width } = Dimensions.get('window');


const IconContainer = styled(TouchableOpacity)`
  width:  ${Size.hamburgerContainerWidth};
  height: ${Size.hamburgerContainerHeigth};
  margin-left: -15px;
  justify-content: center;
  align-items: center;
  zIndex: 1;
  background-color: transparent;
`;

// const toggleDrawer = () => Actions.refresh({ key: 'moiDrawer', open: value => !value });

const ProfileButton = () => <ProfileImageButton />; //eslint-disable-line
export const HamburgerButton = () => (
  <IconContainer onPress={() => {
      store.dispatch(sideActions.sideMenuOpen());
    }}>
    <Icon.Ionicons color="#FFF" name="md-menu" size={Size.hamburgerSize} />
  </IconContainer>
); //eslint-disable-line

export default {
  navigationBarStyle: {
    backgroundColor: 'transparent',
    borderBottomWidth: 0,
  },

  rightButtonStyle: {
    top: Size.mavbarTopSpace - 6,
    position: 'absolute',
    right: 5,
  },

  titleStyle: {
    color: '#FFF',
    fontSize: Size.fontHeader,
    width: width < 320 ? 150 : 180,
    fontFamily: Font.museo('bolder'),
    ...Platform.select({
      ios: {
        top: Size.mavbarTopSpace - 25 ,
      },
      android: {
        top: -10,
      },
    }),
  },

  leftButtonStyle: {
    top: Size.mavbarTopSpace - 7,
    left: 5,
  },

  renderLeftButton: HamburgerButton,
  renderRightButton: ProfileButton,
};
