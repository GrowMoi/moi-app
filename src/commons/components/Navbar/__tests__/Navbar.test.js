import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import 'jest-styled-components/native';
import Navbar from '../Navbar';

const initialState = {
  dimensions: {
    width: 320,
    height: 568,
    orientation: 'PORTRAIT',
  },
};

describe('Navbar', () => {
  it('it should render as expected', () => {
    const wrapper = shallow(<Navbar device={initialState}/>);

    const render = wrapper.dive();
    expect(render).toMatchSnapshot();
  });
});
