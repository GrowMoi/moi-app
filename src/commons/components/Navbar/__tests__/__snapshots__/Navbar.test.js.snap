// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar it should render as expected 1`] = `
<View
  style={
    Array [
      Object {
        "height": 45,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": -2,
      },
      undefined,
    ]
  }
>
  <StatusBar
    animated={false}
    hidden={true}
    showHideTransition="fade"
  />
  <Styled(Image)
    resizeMode="stretch"
    source={1}
    width={320}
  />
</View>
`;
